src/main/scala/io/iohk/ethereum/jsonrpc/EthJsonMethodsImplicits.scala

```diff
@@ -25,18 +25,32 @@ object EthJsonMethodsImplicits extends JsonMethodsImplicits {
   }
 
   implicit val eth_submitHashrate = new JsonDecoder[SubmitHashRateRequest] with JsonEncoder[SubmitHashRateResponse] {
+    val lengthWithPrefix = 66
+
     override def decodeJson(params: Option[JsonAST.JArray]): Either[JsonRpcError, SubmitHashRateRequest] = params match {
-      case Some(JArray(hashRate :: JString(id) :: Nil)) =>
-        extractQuantity(hashRate)
-          .flatMap(h => extractBytes(id).map(i => (h, i)))
-          .map { case (h, i) => SubmitHashRateRequest(h, i) }
+      case Some(JArray((hashRate: JString) :: (id: JString) :: Nil))
+        if hashRate.s.length == lengthWithPrefix && id.s.length == lengthWithPrefix =>
+        val result: Either[JsonRpcError, SubmitHashRateRequest] = for {
+          rate <- extractQuantity(hashRate)
+          miner <- extractBytes(id)
```

Comment:
`extractHash` for `id` will check the length of `id`. `lengthWithPrefix` check is unnecessary.